doc/man7/openssl-core_names.h.pod

```diff
@@ -0,0 +1,49 @@
+=pod
+
+=head1 NAME
+
+openssl/core_names.h - OpenSSL provider parameter names
+
+=head1 SYNOPSIS
+
+ #include <openssl/core_names.h>
+
+=head1 DESCRIPTION
+
+The F<< <openssl/core_names.h> >> header defines a multitude of macros
+for L<OSSL_PARAM(3)> names, algorithm names and other known names used
+with OpenSSL's providers, made available for practical purposes only.
+
+Existing names are further described in the manuals for OpenSSL's
+providers (see L</SEE ALSO>) and the manuals for each algorithm they
+provide (listed in those provider manuals).
+
+=head1 SEE ALSO
+
+L<OSSL_PROVIDER-default(7)>, L<OSSL_PROVIDER-FIPS(7)>
```

Comment:
legacy and NULL also?

Reply:
The null provider doesn't provide much, so I don't see why it should be mentioned here.  I've added the legacy provider to the list, though.